### PR TITLE
WIP: Eliminate the need for inheritance for Admin controllers

### PR DESCRIPTION
**Pull request by lbajsarowicz**
Please do not comment changes yet.
image
Wait for `WIP:` prefix to disappear from the **PR** title!
Thank you.

### Description (*)
Continuation of https://github.com/magento/magento2/pull/26778 to provide backwards-compatible mechanism of ActionInterface for Backend (Adminhtml) actions.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Backend/App/AbstractAction.php [hunk 1]
```diff
@@ -20,31 +20,16 @@
 /**
  * Generic backend controller
  *
- * @deprecated Use \Magento\Framework\App\ActionInterface
+ * @deprecated Use \Magento\Backend\App\ActionInterface
```
Comment (thomas-kl1) [inline app/code/Magento/Backend/App/AbstractAction.php]: Namespace is Framework, not Backend

**Comment by lbajsarowicz** [comment]
**Change of approach**: We do not need to introduce new `ActionInterface` only for Backend. The isolation of directory and separate `di.xml` for `Adminhtml` allows me to use the same `ActionInterface`.

**Comment by lbajsarowicz** [comment]
Suspended for Architectural discussion according to `isAllowed` handling without `AbstractAction`

**Comment by lbajsarowicz** [comment]
Due to limited time and too many architectural concerns, I'm not able to accomplish that before 2.4
Maybe 2.5 would be good time for this change. In the meantime, I'm going to move plugins and observers one by one to the right interfaces.

**Pull request closed without merging**